corehq/util/doc_processor/interface.py

```diff
@@ -131,14 +130,8 @@ def _process_doc(self, doc):
         ok = self.doc_processor.process_doc(doc)
         if ok:
             self.progress.add()
-        else:
-            try:
-                self.document_iterator.retry(doc['_id'], self.max_retry)
-            except TooManyRetries:
-                if not self.doc_processor.handle_skip(doc):
-                    raise
-                else:
-                    self.progress.skip(doc)
+        elif self.doc_processor.handle_skip(doc):
+            self.progress.skip(doc)
```

Comment:
previously this would raise an error if the `handle_skip` returned `False` but now it doesn't. Is that an issue?

Reply:
https://github.com/dimagi/commcare-hq/pull/29743/commits/ffc5ebc65f49751e1b295205088dc02b6c558613